packages/driver/cypress/integration/e2e/multi-domain/multi_domain_cypress_api.spec.ts

```diff
@@ -0,0 +1,253 @@
+// @ts-ignore / session support is needed for visiting about:blank between tests+
+describe('multi-domain Cypress API', { experimentalSessionSupport: true, experimentalMultiDomain: true }, () => {
+  beforeEach(() => {
+    cy.visit('/fixtures/multi-domain.html')
+    cy.get('a[data-cy="multi-domain-secondary-link"]').click()
+  })
+
+  context('Commands', () => {
+    context('add', () => {
+      it('adds a custom command', () => {
+        cy.switchToDomain('foobar.com', () => {
+          // @ts-ignore
+          Cypress.Commands.add('foo', () => {
+            return cy.wrap('bar')
+          })
+
+          // @ts-ignore
+          cy.foo().then((shouldBeBar) => {
+            expect(shouldBeBar).to.equal('bar')
+          })
+        })
+      })
+
+      it('persists defined commands through spec bridge', () => {
```

Comment:
Maybe add a FIXME comment that this should be converted to use 2 switchToDomains in a single test once that's properly supported. I know for some tests we can't get around it since we're testing the tests, but as much as possible, it would be good if tests don't rely on execution order.

Reply:
I think these would technically work right now in once test, but I don't know if it's a good idea to introduce it yet. At a minimum I will add the FIXME comment at the top of the command context to make sure we address it!

Reply:
addressed in ddf1daf3bf